internal/controllers/gateway/gateway_controller.go

```diff
@@ -440,6 +441,11 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		debug(log, gateway, "Ensured gateway was removed from the data-plane (if ever present)")
 		return ctrl.Result{}, nil
 	}
+	err := r.DataplaneClient.UpdateObject(gwc)
```

Comment:
REVIEW: Should we store the gatewayclass into translator cache here or in the gateway class controller?

Reply:
Done. Moved the storing to gwc controller.